feat(docs): add description of `safety`-related `tact.config.json` options to gas and security best practice pages

novusnota opened the pull request:
## Issue

Closes #3204.

novusnota committed:
feat(docs): add description of `safety`-related `tact.config.json` options to gas and security best practice pages

novusnota committed:
changelog

anton-trunov commented on docs/src/content/docs/book/gas-best-practices.mdx:
```diff
@@ -120,6 +120,41 @@ Some functions in the Tact documentation are annotated with a special badge, "50
 
 If you use one of those functions, consider finding cheaper alternatives.
 
+### Pay attention to `safety` option set of `tact.config.json`
```
recommendations should be concrete:

```suggestion
### Disable run-time null checks for well-tested contracts
```

also, we need to mention in the text that we recommend doing this for well-tested contracts only

novusnota replied:
Good suggestion!

novusnota committed:
Apply suggestions from the code review

1. Made the "well-tested contracts" recommendation stand out
2. Moved the section after the "Asm functions" recommendation, as to
visually hint that this is among the last steps that can be taken
